### PR TITLE
Make development experience better

### PR DESCRIPTION
**Pull request by nickbrowne**
Nothing major, just cleaning out some cobwebs to make it less annoying to get a working development environment to suit our needs more.

The things to try are just making sure you can:
```
npm install
createdb sharejs_example
./bin/sharejs
```

And at that point you should have a running ShareJS server backed by postgres and configured with websockets.

### CODE REVIEW

**Comment by nickbrowne** [comment]
Note that the automated build on travis is not currently working, will fix that separately.

Tests are passing, which you can verify yourself if you like:
```
cake --verbose test
```

```
OK: 11903 assertions (36647ms)
All tests succeeded!
```

**Comment by tiagoamaro** [comment]
@nickbrowne, is this ready to be reviewed?

**Comment by nickbrowne** [comment]
Will be once https://github.com/conversation/ShareJS/pull/39 merges, decided to fix the build first.

File: bin/options.js [hunk 1]
```diff
@@ -1,62 +1,30 @@
...
 
 	// Database options
 	db: {
-		// DB type. Options are 'redis', 'couchdb' or 'none'. 'redis' requires the
-		// redis npm package.
+		// DB type. Options are 'postgres' or 'none'.
     //
     // If you don't want a database, you can also say db: null. With no database,
     // all documents are deleted when the server restarts.
+    type: 'pg',
```
Comment (nickbrowne) [inline bin/options.js:12]: One thing worth noting which is not immediately obvious from this change, is that now the tests will run against postgres (previously they were not).

```
✔ db - pg - create doesnt send an error when a document is created
✔ db - pg - create sends an error if the document already exists
✔ db - pg - Calling create multiple times only results in one successful create
✔ db - pg - getSnapshot sends an error for a nonexistant doc
✔ db - pg - getSnapshot has the type, version and snapshot set when a doc is created
✔ db - pg - getSnapshot is updated after calling writeSnapshot()
✔ db - pg - getSnapshot is updated after calling writeSnapshot() with dbMeta
✔ db - pg - getSnapshot returns dbMeta through the callback
✔ db - pg - writeOp writes ops to the DB
✔ db - pg - snapshot is updated when a second op is applied
✔ db - pg - delete a non-existant document sends an error to its callback
✔ db - pg - delete deletes a document
✔ db - pg - delete with dbMeta set still works
✔ db - pg - Calling delete multiple times only results in one successful delete
✔ db - pg - delete with no callback doesnt crash
✔ db - pg - getOps returns [] for a nonexistant document, with any arguments
✔ db - pg - getOps returns [] for a new document, with any arguments
✔ db - pg - getOps returns ops
```

etc

**Comment by nickbrowne** [comment]
This should be good to review again.

**Review by tiagoamaro — APPROVED** [review]
LGTM!

README is now easier to navigate through, removing `cake` command references and assuming more opinionated default configuration values.

**Pull request merged**